services/lab-service/lab/service/topology.py

```diff
@@ -94,10 +101,15 @@ def add_controller(self, controller, batch=True):
         self.vscmd.extend(cmd) if batch else vsctl(cmd)
 
 
-class ASwitch(Switch):
-    def setup(self, a_mappings):
-        self.add_route_flows(a_mappings)
+class KildaSwitch(Switch):
+    @classmethod
+    def create(cls, sw_def):
+        sw = Switch.create(sw_def)
+        sw.define_ports(range(1, sw_def['max_port'] + 1))
+        return sw
+
```

Comment:
ouch. there are 2 options implement `constructor` for that __init__()
or create a classmethod in Switch class

Reply:
updated